### PR TITLE
Silences linker warnings with new build system

### PR DESCRIPTION
**Pull request by flovilmart**
- Remove foundation from linking stems
- Removes system frameworks linking on static lib targets
- For now, gated with environment variable

### CODE REVIEW

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* Silences linker warnings with new build system  
  [flovilmart](https://github.com/flovilmart)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by segiddins** [comment]
This will break any libraries that rely on those flags to be able to import headers modules for those frameworks.

I also think this warning will be eliminated from a future version of the new build system.

**Comment by flovilmart** [comment]
I’ve run it successfully in our large project, and there’s no issue so far. Do you have an example of what would break? I’ll happily investigate.

As Xcode 9.3 / swift 4.1 has been released, I’m not sure we should expect a change any time soon. Perhaps we should report an issue on the swift bug tracker.

Note: it only affects static libs, not frameworks. Dynamically linked targets do not warn.

**Comment by segiddins** [comment]
Once https://github.com/CocoaPods/CocoaPods/pull/7451 is un-reverted, the `-framework` flag will be necessary for the compiler to search those frameworks' `Headers` directories (I'm pretty sure)

File: lib/cocoapods/installer/xcode/pods_project_generator/target_installer.rb [hunk 1]
```diff
@@ -17,6 +17,11 @@ class TargetInstaller
           #
           attr_reader :target
 
+          # @return [Boolean] use_new_build_system
+          #         Wether we're currently using the new build system
+          #
+          attr_accessor :use_new_build_system
```
Comment (dnkoutso) [inline lib/cocoapods/installer/xcode/pods_project_generator/target_installer.rb]: please make this an `attr_reader`

Comment (dnkoutso) [reply lib/cocoapods/installer/xcode/pods_project_generator/target_installer.rb]: Can also set it as `false` to default so you do not have to change all the tests

**Comment by segiddins** [comment]
@flovilmart after thinking this over, I'm moving the build settings changes over to https://github.com/CocoaPods/CocoaPods/pull/7631, and will open a PR to Xcodeproj to stop adding foundation / cocoa when creating static library targets

**Comment by segiddins** [comment]
@flovilmart after thinking this over, I'm moving the build settings changes over to https://github.com/CocoaPods/CocoaPods/pull/7631, and will open a PR to Xcodeproj to stop adding foundation / cocoa when creating static library targets

**Comment by flovilmart** [comment]
This makes sense!

**Comment by dnkoutso** [comment]
Is this fixed by https://github.com/CocoaPods/Xcodeproj/pull/571?

**Comment by segiddins** [comment]
It _should_ be, except for static frameworks

**Comment by Whirlwind** [comment]
Should the resources be coped into the static library?
I think that `Copy Pods Resources` should not integrated if the host target is a Static/Dynamically Framework/Library.

**Pull request closed without merging**